code/messaging/src/phone/java/com/adobe/marketing/mobile/OffersConfig.java

```diff
@@ -0,0 +1,76 @@
+/*
+  Copyright 2022 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile;
+
+import static com.adobe.marketing.mobile.MessagingConstants.LOG_TAG;
+
+import android.app.Application;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageManager;
+
+import com.adobe.marketing.mobile.messaging.BuildConfig;
+
+/**
+ * This class determines which configuration is in use for retrieving offers.
+ */
+final class OffersConfig {
+    private final static String SELF_TAG = "OffersConfig";
+    String activityId;
+    String placementId;
+    String applicationId;
+
+    OffersConfig() {
+        // for E2E functional test use the specified placement and activity id
+        if (BuildConfig.IS_E2E_TEST.get()) {
```

Comment:
Can we pass the activity and placement Id as argument in constructor? Then we won't need to set Boolean fields in Buildconfig.

Reply:
the OffersConfig object is created at the time the InAppNotificationHandler is instantiated. In the case of the functional tests, I wouldn't have direct access to setting the config prior to running the test.